### ISSUE TITLE
AssertionError when combining --configure and --dry-run

### ISSUE THREAD

**Issue by zimmerrol** [hartwork/git-delete-merged-branches#136]
When calling the tool via `git-delete-merged-branches --configure --effort=3 --dry-run --debug` I get the following error message after selecting completing the configuration steps:

```
Traceback (most recent call last):
  File "...\miniconda3\lib\site-packages\git_delete_merged_branches\_cli.py", line 163, in _inner_main
    _innermost_main(config, messenger, colorize)
  File "...\miniconda3\lib\site-packages\git_delete_merged_branches\_cli.py", line 138, in _innermost_main
    git_config = dmb.ensure_configured(config.force_reconfiguration)
  File "...\miniconda3\lib\site-packages\git_delete_merged_branches\_engine.py", line 154, in ensure_configured
    assert self._is_configured(git_config)
AssertionError
```

Am I doing something wrong or is this a bug?
I'm not sure which additional information will be helpful for debugging this, so please let me know if you need more information. I'm using Windows 11 and powershell core as my terminal.

**Comment by hartwork** [comment]
Hi @zimmerrol,

thanks for bringing this here. During live configuration git-delete-merged-branches adds a key `delete-merged-branches.configured` to `<gitdir>/config` by means of the `git config` command. Its value is supposed to be `5.0.0+` as can be seen here:

```
# git config --get delete-merged-branches.configured
5.0.0+
```

For some reason writing and re-reading that config key and value did not work. Could you run `git config --get delete-merged-branches.configured` on this repository and also `git --version` and share their output? (I have Git 2.41.0 on Linux here at the moment and no easy access to Windows.)

Thanks!

**Comment by zimmerrol** [comment]
Hi @hartwork,
Thanks for getting back to me that quickly. 
It seems like this key has not been set. Is it possible that the `--dry-run` flag disables writing this flag?

```powershell
>> git config --get delete-merged-branches.configured
>> git --version
git version 2.29.2.windows.3
```

**Comment by hartwork** [comment]
Hi @zimmerrol I overlooked the `--dry-run` somehow, that's it, thanks for bringing it to my attention.

I think there are at least these theoretical options:

- a) Make `--dry-run` and live configuration mutually exclusive officially and technically; would need to cover explicit `--configure` and implicit need
- b) Drop/adjust the assertion (after double-checking it's fully safe to)
- c) Pass a cached copy of `git config` state around at/to more places and update that cache also for `--dry-run`
- d) Exclude config changes from `--dry-run` and major bump the version to accompany that change

So far I don't really like any of these options. I'll have to sleep over this. Let me know if you can think of alternatives that I might be missing or how feel about these options, if you feel like it.

**Comment by olejorgenb** [comment]
I wasted some time on this yesterday, and while I think it's technically not a bug, I would prefer (a) - simply print an error message if configuration is triggered when `--dry-run` is active. (c) seems like ton of work for very limited benefit? (implement a "deconfigure" action instead, clearing the config file of any dmd keys?)

**Comment by hartwork** [comment]
Fixed in 7.4.2 now.
